lib/ex_unit/lib/ex_unit/capture_log.ex

```diff
@@ -84,27 +84,21 @@ defmodule ExUnit.CaptureLog do
   end
 
   defp add_capture(pid, opts) do
-    parent = self()
-    {:ok, proxy_pid} = Task.start fn ->
-      case :gen_event.add_sup_handler(Logger, {Console, pid}, {pid, opts}) do
-        :ok ->
-          ref = Process.monitor(parent)
-          send parent, {self(), :ok}
-          receive do
-            {:DOWN, ^ref, :process, ^parent, _reason} -> :ok
-            {:gen_event_EXIT, {Console, ^pid}, _reason} -> :ok
-          end
-        other ->
-          send parent, {self(), other}
-      end
-    end
-    ref = Process.monitor(proxy_pid)
-    receive do
-      {^proxy_pid, result} ->
-        Process.demonitor(ref, [:flush])
-        result
-      {:DOWN, ^ref, :process, ^proxy_pid, reason} ->
-        {:error, reason}
+    :proc_lib.start(__MODULE__, :init_proxy, [pid, opts, self()])
```

Comment:
can we wrap this in a case and do an `exit/1` like in `remove_capture/1` so we get a clearer understanding of the error rather than a MatchError.